code/modules/power/power.dm

```diff
@@ -132,28 +132,100 @@
-	A?.use_power(amount, chan)
+/**
+ * Returns the available energy from the apc's cell and grid that can be used.
+ * Args:
+ * - consider_cell: Whether to count the energy from the APC's cell or not.
+ * Returns: The available energy the machine can access from the APC.
+ */
+/obj/machinery/proc/available_energy(consider_cell = TRUE)
+	var/area/home = get_area(src)
+
+	if(!home)
+		return FALSE
+	if(!home.requires_power)
+		return INFINITY
+
+	var/obj/machinery/power/apc/local_apc = home.apc
+	if(!local_apc)
+		return FALSE
+
+	return consider_cell ? local_apc.available_energy() : local_apc.surplus()
 
 /**
- * An alternative to 'use_power', this proc directly costs the APC in direct charge, as opposed to being calculated periodically.
- * - Amount: How much power the APC's cell is to be costed.
+ * Draws energy from the APC. Will use excess energy from the APC's connected grid,
+ * then use energy from the APC's cell if there wasn't enough energy from the grid, unless ignore_apc is true.
+ * Args:
+ * - amount: The amount of energy to use.
+ * - channel: The power channel to use.
+ * - ignore_apc: If true, do not consider the APC's cell when demanding energy.
+ * - force: If true and if there isn't enough energy, consume the remaining energy. Returns 0 if false and there isn't enough energy.
+ * Returns: The amount of energy used.
  */
-/obj/machinery/proc/directly_use_power(amount)
+/obj/machinery/proc/use_energy(amount, channel = power_channel, ignore_apc = FALSE, force = TRUE)
+	if(amount <= 0) //just in case
+		return FALSE
+	var/area/home = get_area(src)
+
+	if(!home)
+		return FALSE //apparently space isn't an area
+	if(!home.requires_power)
+		return amount //Shuttles get free power, don't ask why
+
+	var/obj/machinery/power/apc/local_apc = home.apc
+	if(!local_apc)
+		return FALSE
+
+	// Surplus from the grid.
+	var/surplus = local_apc.surplus()
+	var/grid_used = min(surplus, amount)
+	var/apc_used = 0
+	if((amount > grid_used) && !ignore_apc) // Use from the APC's cell if there isn't enough energy from the grid.
+		apc_used = local_apc.cell.use(amount - grid_used, force = force)
+
+	if(!force && (amount < grid_used + apc_used)) // If we aren't forcing it and there isn't enough energy to supply demand, return nothing.
+		return FALSE
+
+	// Use the grid's and APC's energy.
+	amount = grid_used + apc_used
+	local_apc.add_load(grid_used JOULES)
+	home.use_energy(amount JOULES, channel)
+	return amount
+
+/**
+ * Does what use_energy() does, but it automatically scales the amount by the machine's subsystem wait.
+ * This draws the energy needed such that it will be drawing amount watts if this was called every process().
+ * Args:
+ * - amount: The amount of power to use.
+ * - channel: The power channel to use.
+ * - ignore_apc: If true, do not consider the APC's cell when demanding power.
+ * Returns: The amount of energy used.
+ */
+/obj/machinery/proc/use_power(amount, channel = power_channel, ignore_apc = FALSE, force = TRUE)
```

Comment:
I haven't seen this proc used anywhere till now. Did i miss a use case or when do we need this? because in your PR body you did specify everything draws power from the grid as energy.

Reply:
I guess we don't need this. I just left that there handy in case I got in an odd scenario where I wanted power consumption in some process without access to seconds_per_tick.

Reply:
Well if you have no need for this proc in your PR i suggest you remove it. Writing helper procs for potential future use cases is how dead code is born

Reply:
Removed.